### PR TITLE
DAOS-3315 drpc: statically alloc resp in drpc handlers

### PR DESCRIPTION
**Pull request by tanabarr**
Statically allocate protobuf response objects to reduce boilerplate
related to dynamic allocation.

Other miscellanea:
* remove erroneous comment noticed in unrelated review
* update drpc status failure code

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: src/mgmt/srv_drpc.c [hunk 1]
```diff
@@ -252,43 +195,31 @@ ds_mgmt_drpc_create_mgmt_svc(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
...
-	}
-
-	/* Response status is populated with SUCCESS on init. */
-	mgmt__daos_resp__init(resp);
-
 	rc = ds_mgmt_svc_start(false /* !create */, 0 /* size */,
 			       false /* !bootstrap */, NULL /* uuid */,
 			       NULL /* addr */);
-	if (rc == -DER_ALREADY) {
+	if (rc == -DER_ALREADY)
```
Comment (kjacque) [inline src/mgmt/srv_drpc.c]: I think in this case we should set rc = 0, since there was nothing to do. The old logic only set rc in the other error case.

Comment (tanabarr) [reply src/mgmt/srv_drpc.c]: good catch thanks

File: src/mgmt/srv_drpc.c [hunk 2]
```diff
@@ -205,10 +205,11 @@ ds_mgmt_drpc_start_mgmt_svc(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 			       NULL /* addr */);
 	if (rc == -DER_ALREADY)
```
Comment (kjacque) [inline src/mgmt/srv_drpc.c]: style nit - if one clause of the if statement has curly brackets, both should.

Comment (tanabarr) [reply src/mgmt/srv_drpc.c]: yes was considering changing back

**Pull request merged**